src/python/pants/engine/README.md

```diff
@@ -46,15 +46,15 @@ As a very simple example, you might register the following `@rule` that can comp
 Product given a single `Int` input.
 
 ```python
-@rule(StringType, [Select(IntType)])
+@rule(StringType, [IntType])
 def int_to_str(an_int):
   return '{}'.format(an_int)
 ```
 
 The first argument to the `@rule` decorator is the Product (ie, return) type for the `@rule`. The
-second argument is a list of `Selector`s that declare the types of the input arguments to the
-`@rule`. In this case, because the Product type is `StringType` and there is one `Selector`
-(`Select(IntType)`), this `@rule` represents a conversion from `IntType` to `StrType`, with no
+second argument is a list of parameter selectors that declare the types of the input parameters for
+the `@rule`. In this case, because the Product type is `StringType` and there is one parameter
+selector (for `IntType`), this `@rule` represents a conversion from `IntType` to `StrType`, with no
```

Comment:
"parameter selectors" is a very good edit, this leaves the opportunity to expand it later as needed while making it extremely clear what those types are doing there.